### DISCUSSION TITLE
How to push updated ".env" files that are not tracked to remote repositories?

### DISCUSSION THREAD

**Discussion by dushmanta05** [community/community#63903] [Repositories]
Hi! I have a project on GitHub, and this project requires an API key to fetch data from the servers. Since I'm using *vite.js* for bundling, I've made a **.env** file specifying ***VITE_KEY*** with the API key as its value (as mentioned in the Vitejs docs), and I've added the **.env** file to the **.ignore** file so that it remains private. Now I've updated the API key with a new one, and I'm unable to update the remote repo from local. Can anyone help? How can I do this? Also previously, the website was running smoothly, but since the API limit was exceeded, I had to change it.

**Comment by nirav-gajera** [comment]
1. Make sure that the new API key is valid and that you have the correct permissions to use it.
2. Check if the .env file is added to the .gitignore file. If it is, remove the .env file from the .gitignore file.
3. Commit and push your changes to the remote repo:

```
git add .env
git commit -m "Add new API key"
git push origin main
```

**Reply by dushmanta05** [reply]
Hi! Thank you for your response.

Don't you think removing the **.env** file from the **.ignore** file will expose my API keys when I push that to the remote repository?
